api/preprints/serializers.py

```diff
@@ -287,6 +299,60 @@ def update(self, preprint, validated_data):
             preprint.original_publication_date = validated_data['original_publication_date'] or None
             save_preprint = True
 
+        if 'has_prereg_links' in validated_data:
+            if not switch_is_active(SLOAN_STUDY_PREREG):
+                raise exceptions.ValidationError(
+                    detail='You do not have ability to edit your prereg link availability at this time.',
+                )
+
+            has_prereg_links = validated_data['has_prereg_links']
+            preprint.update_has_prereg_links(auth, has_prereg_links)
+
+        if 'why_no_prereg' in validated_data:
+            if not switch_is_active(SLOAN_STUDY_PREREG):
+                raise exceptions.ValidationError(
+                    detail='You do not have ability to edit your prereg link availability at this time.',
```

Comment:
This is a lot of the same kind of code. I feel like this could be DRYed. Maybe a conditional field composition for waffle switches/flags? It could either not show the field or could make the field read only in the case of the waffle switch not being active.

For the ones with more complex validation, those might make more sense to use [validators](https://www.django-rest-framework.org/api-guide/validators/).